app/views/session_timeout/_ping.js.erb

```diff
@@ -26,12 +30,12 @@ function success(data) {
       time_cutoff = new Date().getTime() + warning,
       show_warning = time_timeout < time_cutoff;
 
-  if (show_warning & !el) {
-    cntnr.insertAdjacentHTML('afterbegin', warning_info);
+  if (show_warning & !modal.shown) {
```

Comment:
Happy coincidence that the bitwise works, but that seems brittle. Yes please change. Would be interesting to see if tests all work the same afterwards.